lib/spack/spack/test/util/compression.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+import shutil
+
+import pytest
+
+from llnl.util.filesystem import working_dir
+
+from spack.paths import spack_root
+from spack.util import compression as scomp
+from spack.util.executable import CommandNotFoundError
+
+datadir = os.path.join(spack_root, 'lib', 'spack',
+                       'spack', 'test', 'data', 'compression')
+
+ext_archive = {}
+[ext_archive.update({ext: '.'.join(['Foo', ext])}) for
+    ext in scomp.ALLOWED_ARCHIVE_TYPES]
+
+
+def support_stub():
+    return False
+
+
+@pytest.fixture
+def compr_support_check(monkeypatch):
+    monkeypatch.setattr(scomp, 'lzma_support', support_stub)
+    monkeypatch.setattr(scomp, 'tar_support', support_stub)
+    monkeypatch.setattr(scomp, 'gzip_support', support_stub)
+    monkeypatch.setattr(scomp, 'bz2_support', support_stub)
+
+
+@pytest.fixture
+def archive_file(tmpdir_factory, request):
+    """Copy example archive to temp directory for test"""
+    archive_file_stub = os.path.join(datadir, 'Foo')
+    extension = request.param
+    tmpdir = tmpdir_factory.mktemp('compression')
+    shutil.copy(archive_file_stub + '.' + extension, tmpdir)
+    return os.path.join(str(tmpdir), 'Foo.%s' % extension)
+
+
+@pytest.mark.parametrize('archive_file', scomp.ALLOWED_ARCHIVE_TYPES, indirect=True)
+def test_native_unpacking(tmpdir_factory, archive_file):
+    extension = scomp.extension(archive_file)
+    util = scomp.decompressor_for(archive_file, extension)
+    tmpdir = tmpdir_factory.mktemp("comp_test")
+    with working_dir(str(tmpdir)):
+        assert not os.listdir(os.getcwd())
+        util(archive_file)
+        assert os.listdir(os.getcwd())
```

Comment:
What are the contents of the Foo archive? It occurred to me it would be slightly more robust to assert for example that the archive contents are what is expected (e.g. if there's a file in the archive that contains "text", read the file and check it) - that isn't essential but describing the contents would be useful IMO.

Reply:
Currently it's an empty extension less file with the name `Foo`, so I would have to regenerate all the archives with a file with contents. Not a problem, just letting you know that it may extend the life of this PR if we want to go that way. IMO we may as well while we're here.

Reply:
Ah OK apologies for not fleshing out that idea, but yes I think it would be great to place some content in the archive.

* The tar/zip ones can be a directory with a file that just has one word
* The gunzip/bunzip2/etc. ones can be a file with one word